lib/matplotlib/backend_bases.py

```diff
@@ -2150,21 +2073,30 @@ def get_supported_filetypes_grouped(cls):
             groupings[name].sort()
         return groupings
 
-    def _get_output_canvas(self, backend, fmt):
+    @contextmanager
+    def _switching_canvas_and_get_print_method(self, fmt, backend=None):
```

Comment:
The verb tenses appear mixed up in the name here.

Reply:
I realize the name is a bit weird, do you have a better suggestion?  (the canvas is switched for the duration of the cm; the print_method immediately yielded)

Reply:
```suggestion
    def _switch_canvas_and_return_print_method(self, fmt, backend=None):
```
?  Still not great, but the tense reads a little better to me.  Thought about tacking on `_CM` but that made it worse..

Reply:
Works for me, renamed it.